pingrid.py

```diff
@@ -448,13 +452,20 @@ def parse_colormap(s: str) -> np.ndarray:
     #     len(vs),
     #     [f"{v.red:02x}{v.green:02x}{v.blue:02x}{v.alpha:02x}" for v in vs],
     # )
-    rs = np.array([vs[int(i / 256.0 * len(vs))] for i in range(0, 256)], np.uint8)
+    if thresholds is None:
+        rs = np.array([vs[int(i / 256.0 * len(vs))] for i in range(0, 256)], np.uint8)
+    else:
+        rs = np.full([256, 4], np.nan)
+        for i in range(0, thresholds.size):
+            rs[int(255.*(thresholds[i]-thresholds[0])/(thresholds[-1]-thresholds[0])), :] = np.array([vs[i]], np.uint8)
+        dfrs = pd.DataFrame(rs)
```

Comment:
We've mostly been using the convention of suffixes `_df`, `_da`, and `_ds` for DataFrame, DataArray, and DataSet. Following that convention, this would be called `rs_df`.